torch/nn/functional.py

```diff
@@ -2391,16 +2511,26 @@ def mish(input: Tensor, inplace: bool = False) -> Tensor:
     .. note::
         See `Mish: A Self Regularized Non-Monotonic Neural Activation Function <https://arxiv.org/abs/1908.08681>`_
 
+    Keyword args:
+        out: the output tensor. If :attr:`out` is used, this operation won't
+            be differentiable. Default: None
+
     See :class:`~torch.nn.Mish` for more details.
     """
     if has_torch_function_unary(input):
-        return handle_torch_function(mish, (input,), input, inplace=inplace)
+        return handle_torch_function(mish, (input,), input, inplace=inplace, out=out)
+
+    if inplace and out is not None:
+        raise ValueError(
+            "The parameters `inplace` and `out` are mutually exclusive."
+            f" You provided {inplace=}, {out=}")
+
     if inplace:
         return torch._C._nn.mish_(input)
-    return torch._C._nn.mish(input)
+    return torch._C._nn.mish(input, out=out)
 
 
-def hardswish(input: Tensor, inplace: bool = False) -> Tensor:
+def hardswish(input: Tensor, inplace: bool = False, *, out: Optional[Tensor] = None) -> Tensor:
```

Comment:
@bdhirsh 

Looks like this change is not compatible with `torch.jit`:

https://github.com/pytorch/pytorch/actions/runs/10351921869/job/28679957034

```shell
24-08-12T23:56:24.6828471Z Module 'M' has no attribute 'activation_fn' (This function exists as an attribute on the Python module, but we failed to compile it to a TorchScript function. 
2024-08-12T23:56:24.6829542Z The error stack is reproduced here:
2024-08-12T23:56:24.6830398Z Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults:
2024-08-12T23:56:24.6831580Z   File "/opt/conda/envs/py_3.11/lib/python3.11/site-packages/torch/nn/functional.py", line 1796
2024-08-12T23:56:24.6832307Z     inplace: bool = False,
2024-08-12T23:56:24.6832624Z     *,
2024-08-12T23:56:24.6832882Z     out: Optional[Tensor] = None
2024-08-12T23:56:24.6833313Z                             ~~~~ <--- HERE
2024-08-12T23:56:24.6833762Z ) -> Tensor:  # noqa: D400,D402
```

And not restricting `out` to be keyword only presents other problems, for example for `hardtanh`:

```python
RuntimeError:
Module 'M' has no attribute 'activation_fn' (This function exists as an attribute on the Python module, but we failed to compile it to a TorchScript function.
The error stack is reproduced here:

Arguments for call are not valid.
The following variants are available:

  aten::hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor:
  Expected at most 3 arguments but found 4 positional arguments.

  aten::hardtanh.out(Tensor self, Scalar min_val=-1, Scalar max_val=1, *, Tensor(a!) out) -> Tensor(a!):
  Argument out not provided.

The original call is:
  File "/working/pytorch_fork/pytorch-fork/torch/nn/functional.py", line 1813
    if inplace:
        return torch._C._nn.hardtanh_(input, min_val, max_val)
    return torch._C._nn.hardtanh(input, min_val, max_val, out)
           ~~~~~~~~~~~~~~~~~~~~~ <--- HERE

```

And since this [ticket](https://github.com/pytorch/pytorch/issues/47565)  is still open (and `torch.jit` is deprecated?) I guess there isn't a way that `torch.nn.functional` can receive `out` arguments?

Reply:
oh. That is unfortunate - but yes, I don't think this is likely to be fixed on the `torch.jit` side, and we probably don't want to break any existing users of jit